salalib/geometrygenerators.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2017 Petros Koutsolampros
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "geometrygenerators.h"
+
+std::vector<Point2f> GeometryGenerators::generateDiskTriangles(int sides, float radius, Point2f position) {
+    std::vector<Point2f> diskTriangles;
+    for(int i = 0; i < sides; i++) {
+        diskTriangles.push_back(Point2f(position.x, position.y));
+        diskTriangles.push_back(Point2f(position.x + radius*sin(2*M_PI*(i+1)/sides), position.y + radius*cos(2*M_PI*(i+1)/sides)));
+        diskTriangles.push_back(Point2f(position.x + radius*sin(2*M_PI*i/sides), position.y + radius*cos(2*M_PI*i/sides)));
+    }
+    return diskTriangles;
+}
+
+std::vector<Point2f> GeometryGenerators::generateMultipleDiskTriangles(int sides, float radius, std::vector<Point2f> positions) {
+    const std::vector<Point2f> &diskTriangles = generateDiskTriangles(sides , radius);
```

Comment:
I am pretty sure this shouldn't be a const reference - you should actually own the copy on the stack